independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/RequestContext.java

```diff
@@ -251,7 +264,7 @@ static class RequestContextState implements ContextState {
 
         RequestContextState(ConcurrentMap<Contextual<?>, ContextInstanceHandle<?>> value) {
             this.map = Objects.requireNonNull(value);
-            this.isValid = 1;
+            assert isValid == VALID;
```

Comment:
Validation of the `assert` keyword is not always enabled, or?

Reply:
Nope, it should be enabled just on tests (with -ea), but I can removed it too; it's just to be sure that initial value won't change (like a self doc check)

Reply:
I think adding a comment would be enough, just to remind that the field is initialized to the default value of the type, which is `0`.

Reply:
Definitely +1 for a comment

Reply:
I'll go for it then, thanks!

Reply:
Nope, not just for tests: I mean that assertions should be enabled there by default AFAIK, but is more a check for outself of the future in case we change the init semantic. But as said by others, comments can do a much better job then this, for humans

Reply:
+1 for a comment ;-)

Reply:
@mkouba how it looks like now? I've added a comment re the old JLS on java 8 as well, for completeness